docs/source/architecture/data_services_architecture.rst

```diff
@@ -1,4 +1,110 @@
+
+Added Jdo Imports are as follows:-
+javax.jdo,javax.jdo.identity,javax.jdo.spi,org.joda.time,org.apache.commons.lang,org.springframework.transaction.support,org.motechproject.mds.filter,org.motechproject.mds.query,org.motechproject.mds.util,org.motechproject.commons.date.util,org.datanucleus.enhancer,org.datanucleus,org.datanucleus.state,org.datanucleus.asm,org.datanucleus.exceptions,org.datanucleus.metadata,org.datanucleus.plugin,org.datanucleus.util,org.datanucleus.identity
+
+
+#. MDS Constructor: It provides methods to create entities, their service implementations and corresponding interfaces.
+
+#. Entity Builder: It is use to build classes of a given entities.
+
+#. Infrastructure Builder: It is responsible for building infrastructure for a given entity .i.e repository, interface and service classes. These classes are created only if not available in the classpath using  javassist to construct them.
+
+#. Metadata Builder: It is responsible for generating JDO metadata for entity.
+
+#. Enum Builder: Value defined in Combo Box/ Field Metadata will be used to create an Enum.
+
+#. Lookup Builder: It allows to create a method without body and added into javassist service interface and method with body added into javassist service class.
+
+#. Processors: Processors process DDEs based on annotations and create Infrastructure and Metadata Information for those entities which can used via DataNucleus enhancer to enhance entities to use them for Persistence Operations.
+
+#. JDO Enhancer: It extends DataNuclues JDOEnhancer. It enhances classes to add missing information so that the classes can be used by DataNuclues for persistance related operations.
+
+#. MDS Data Provider: It creates Mds data provider json to register in task module.
+
+#. Infrastructure Classes, EUDE and DDE: It represents all entities, services, repositories of entities used by MDS Web module to represent MDS entities in UI.
+
+#. Pre Defined Service Section:
+Mds provide some predefined services used by Mds Web and other modules of motech platform :
+
+    *Action Handler Service: It provides methods for handling tasks actions events related
+    with MDS CRUD operations .i.e. create, update and delete.
+
+    * Bundle watcher Suspension Service: An OSGi service, allowing to temporarily disable bundle processing by MDS. When processing gets suspended, MDS will still listen to bundle events, but instead of processing them, they will be queued and processed after the processing gets restored. This allows, besides others, to install/uninstall a larger amount of bundles at one time, without facing annotation processing problems.
```

Comment:
This is a very important component that does a lot of stuff under the covers, such as calling the annotation processor, using the JarGeneratorService to generated the bundle and so on. It should have its own section,